src/components/Uploader/index.js

```diff
@@ -30,64 +38,13 @@ export const UploadError = ({errorMessage}) => (
   <div className={`${style.text} ${style.error}`}>{errorMessage}</div>
 )
 
-export class Uploader extends Component {
-
-  componentDidMount () {
-    this.canvas = document.createElement('canvas')
-  }
-
-  handleUpload = (files) => {
-    const { method, handleImage, setUploadState } = this.props
-    setUploadState(true)
-    const options = {
-      maxWidth: 960,
-      maxHeight: 960,
-      canvas: true
-    }
-    const [ file ] = files
-    loadImage(file.preview, (canvas) => {
-      events.emit('imageLoaded', canvas)
-    }, options)
-    events.once('imageLoaded', (canvas) => {
-      const image = canvas.toDataURL('image/webp')
-      const payload = {
-        id: randomId(),
-        messageType: method,
-        image
-      }
-      return handleImage(method, payload)
-    })
-  }
-
-  renderUploader = (captured) => {
-    if (captured) {
-      return ( <Confirm {...this.props} /> )
-    }
-
-    return this.renderDropzone()
-  }
-
-  renderDropzone = () => {
-    const { uploading, noDocument } = this.props
-    return (
-      <Dropzone
-        onDrop={this.handleUpload}
-        multiple={false}
-        className={style.dropzone}
-      >
-        {uploading && <UploadProcessing /> || <UploadInstructions />}
-        {(!uploading && noDocument) && <DocumentNotFound />}
-      </Dropzone>
-    )
-  }
-
-  render () {
-    const { method, documentCaptured, faceCaptured } = this.props
-    const methods = {
-      'document': () => this.renderUploader(documentCaptured),
-      'face': () => this.renderUploader(faceCaptured),
-      'home': () => null
-    }
-    return (methods[method] || methods['home'])()
-  }
-}
+export const Uploader = ({method, onImageSelected, uploading, noDocument}) => (
+  <Dropzone
+    onDrop={([ file ])=> onImageSelected(file)}
+    multiple={false}
+    className={style.dropzone}
+  >
+    {uploading && <UploadProcessing /> || <UploadInstructions />}
```

Comment:
yeah, I agree, but I was focusing more on splitting pure code from the state code, but those one liners should at some point be changed to something more readable.